### PR TITLE
Feat/timestamps

### PR DESCRIPTION
**Pull request by mikhailswift**
* Adds timestamp metadata to any signatures that may have timestamping information attached.

Also,

Archivist used the Store and Download functions which returned
envelopes, parsed on either end by go's json decoder and encoder. This
could cause a new line to be appended to an envelope after encoding,
causing a difference in the envelope's hash.

By using the Reader/Writer methods archivistctl just passes through the
raw bytes to archivist, allowing the server to decode and verify any
information before passing those raw bytes to object storage. This
ensures gitoids and shasums of envelopes remain consistent.

### CODE REVIEW

**Comment by mikhailswift** [comment]
Sample query to test this:

```
{
  dsses(first:10,
  where:{
    hasSignaturesWith:{
      hasTimestamps: true
    }
  }) {
    edges{
      node{
        gitoidSha256
        signatures{
          keyID
          timestamps{
             timestamp
          }
        }
      }
    }
  }
}
```

example results from above query:

```
{
  "data": {
    "dsses": {
      "edges": [
        {
          "node": {
            "gitoidSha256": "3e2d12258466b2ec5b2c65a87232c8d2356e3532044185f39a4e63b1a77a5777",
            "signatures": [
              {
                "keyID": "ae2dcc989ea9c109a36e8eba5c4bc16d8fafcfe8e1a614164670d50aedacd647",
                "timestamps": [
                  {
                    "timestamp": "2022-11-15T00:06:03Z"
                  }
                ]
              }
            ]
          }
        }
      ]
    }
  }
}
```

Will get this pushed to the dev instance tomorrow.

File: ent.graphql [hunk 1]
```diff
@@ -529,3 +533,53 @@ input SubjectWhereInput {
...
+  typeContainsFold: String
+  """timestamp field predicates"""
+  timestamp: Time
+  timestampNEQ: Time
+  timestampIn: [Time!]
+  timestampNotIn: [Time!]
+  timestampGT: Time
+  timestampGTE: Time
+  timestampLT: Time
+  timestampLTE: Time
```
Comment (colek42) [inline ent.graphql:581]: This will be great!

File: internal/metadatastorage/mysqlstore/mysql.go [hunk 2]
```diff
@@ -150,14 +157,30 @@ func (s *Store) Store(ctx context.Context, gitoid string, obj []byte) error {
...
 				SetKeyID(sig.KeyID).
 				SetSignature(base64.StdEncoding.EncodeToString(sig.Signature)).
 				SetDsse(dsse).
 				Save(ctx)
 			if err != nil {
 				return err
 			}
+
+			for _, timestamp := range sig.Timestamps {
+				timestampedTime, err := timeFromTimestamp(timestamp)
```
Comment (colek42) [inline internal/metadatastorage/mysqlstore/mysql.go:170]: Correct me if I am wrong, but we will only store the time if a timestamp exists.

**Review by colek42 — APPROVED** [review]

**Comment by colek42** [comment]
Great Work!

File: internal/metadatastorage/mysqlstore/mysql.go (hunk 2 above)
Comment (mikhailswift) [reply internal/metadatastorage/mysqlstore/mysql.go:170]: Correct. I’m splitting up the other time stuff currently

**Pull request merged**